webapp/graphite/worker_pool/pool.py

```diff
@@ -1,58 +1,117 @@
 import Queue
+import time
 
 from threading import Lock
 from django.conf import settings
 from multiprocessing.pool import ThreadPool
 
-__init_lock = Lock()
-__pools = {}
+_init_lock = Lock()
+_pools = {}
+
+
+class Job(object):
+  """A job to be executed by a pool
+
+  The job accepts a function and arguments.
+
+  When it is run, it will execute the function with the specified arguments.
+
+  The return value of the function will be stored in the result property of the job.
+
+  If the function raises an exception, it will be stored in the exception property of the job.
+  """
+  __slots__ = ('func', 'args', 'kwargs', 'result', 'exception')
+
+  def __init__(self, func, *args, **kwargs):
+    self.func = func
+    self.args = args
+    self.kwargs = kwargs
+    self.result = None
+    self.exception = None
+
+  def run(self):
+    try:
+      self.result = self.func(*self.args, **self.kwargs)
+    except Exception as err:
+      self.exception = err
```

Comment:
The pool now accepts and returns `Job` objects, which makes it easier to keep track of what's going on, and to deal with functions that raise exceptions.